app/v2/controllers/V2MovementsController.scala

```diff
@@ -533,7 +533,8 @@ class V2MovementsControllerImpl @Inject() (
                  |Reason: ${failureDetails.failureReason}
                  |Message: ${failureDetails.message}""".stripMargin)
 
-            // TODO: Push notification on failure
+            pushNotificationsService
+              .postPpnsNotification(movementId, messageId, Json.toJson(UpscanFailedResponse(reference, failureDetails)))
```

Comment:
What does `UpscanFailedResponse` look like?

Reply:
Wait, no, this isn't appropriate to send.

This should really be a `PresentationError.badResponse` with a message that's appropriate, perhaps something like "Uploaded file not accepted". We should NOT be sending back potentially internal details.